package aminer;

option java_outer_classname = "Mashaler";
option java_package = "org.iminer.services.datacenter";

message EntitySearchRequest {
    required string query = 1;
    optional int32 offset = 2;
    optional int32 count = 3;
}

message EntityDetailRequest {
    repeated int32 id = 1;
}

message Stat {
    required string type = 1;
    required int32 value = 2;
}

message RelatedEntity {
    required string type = 1;
    repeated int32 id = 2;
}

message DetailedEntity {
    required string title = 1;
    required int32 id = 2;
    optional string url = 3;
    optional string description = 4;
    repeated Stat stat = 5;
    optional string topics = 6;
    optional string imgurl = 7;
    optional int32 original_id = 8;
    repeated RelatedEntity related_entity = 9;
}

message EntitySearchResponse {
    required string query = 1;
    required int32 total_count = 2;
    repeated DetailedEntity entity = 3;
}

message Influence {
    required int32 id = 1;
    required int32 topic = 2;
    required double score = 3;
}

message InfluenceSearchResponse {
    required int32 entity_id = 1;
    repeated Influence influence = 2;
    repeated Influence influenced_by = 3;
}

